src/aap_eda/wsapi/consumers.py

```diff
@@ -161,6 +163,9 @@ def handle_heartbeat(self, message: HeartbeatMessage) -> None:
                 message.stats["ruleSetName"]
             ] = message.stats
             activation.save(update_fields=["ruleset_stats"])
+
+            if activation.status == ActivationStatus.STARTING:
+                orchestrator.monitor_rulebook_processes()
```

Comment:
I understand what you said. But I don't think it will cause any conflicts because it is an unique task. What I should do here is to call `enqueue_monitor_rulebook_processes` instead of `monitor_rulebook_process()`. It does not matter it is called by a scheduler or any other code.